src/solver/conv_mlir_igemm_fwd_xdlops.cpp

```diff
@@ -69,16 +70,139 @@ bool ConvMlirIgemmFwdXdlops::IsApplicable(const ConvolutionContext& ctx) const
+                                                               int GemmKPerBlock_,
+                                                               int GemmMPerWave_,
+                                                               int GemmNPerWave_,
+                                                               int GemmKPACKSize_,
+                                                               bool GemmAThreadCopyMoreGemmK_,
+                                                               bool GemmBThreadCopyMoreGemmKPack_,
+                                                               bool use_spare_set_)
+    : GemmMPerBlock(GemmMPerBlock_),
+      GemmNPerBlock(GemmNPerBlock_),
+      GemmKPerBlock(GemmKPerBlock_),
+      GemmMPerWave(GemmMPerWave_),
+      GemmNPerWave(GemmNPerWave_),
+      GemmKPACKSize(GemmKPACKSize_),
+      GemmAThreadCopyMoreGemmK(GemmAThreadCopyMoreGemmK_),
+      GemmBThreadCopyMoreGemmKPack(GemmBThreadCopyMoreGemmKPack_),
+      use_spare_set(use_spare_set_)
+{
+}
+
+PerformanceConvMlirIgemmXdlops::PerformanceConvMlirIgemmXdlops(bool spare)
+    : PerformanceConvMlirIgemmXdlops::PerformanceConvMlirIgemmXdlops(
+          4, 16, 1, 4, 16, 1, false, false, spare)
+{
+}
+
+PerformanceConvMlirIgemmXdlops::PerformanceConvMlirIgemmXdlops()
+    : PerformanceConvMlirIgemmXdlops::PerformanceConvMlirIgemmXdlops(
+          -1, -1, -1, -1, -1, -1, false, false)
+{
+}
+
+bool PerformanceConvMlirIgemmXdlops::operator==(const PerformanceConvMlirIgemmXdlops& other) const
+{
+    // clang-format off
+    return GemmMPerBlock == other.GemmMPerBlock
+        && GemmNPerBlock == other.GemmNPerBlock
+        && GemmKPerBlock == other.GemmKPerBlock
+        && GemmMPerWave == other.GemmMPerWave
+        && GemmNPerWave == other.GemmNPerWave
+        && GemmKPACKSize == other.GemmKPACKSize
+        && GemmAThreadCopyMoreGemmK  == other.GemmAThreadCopyMoreGemmK
+        && GemmBThreadCopyMoreGemmKPack  == other.GemmBThreadCopyMoreGemmKPack
+        && use_spare_set == other.use_spare_set;
+    // clang-format on
+}
+
+bool PerformanceConvMlirIgemmXdlops::IsValid(const ConvolutionContext& ctx) const
+{
+#if MIOPEN_USE_MLIR
+    bool isValid = MiirIsConfigApplicable(
+        mlir::ConstructBuildOptions(ctx, GetOperation(), GetKernelName(), ToString(), true));
+    return isValid;
+#else
+    std::ignore = ctx;
+    return false;
+#endif
+}
+
+bool PerformanceConvMlirIgemmXdlops::SetNextValue(const ConvolutionContext& /*config*/)
+{
+    GemmBThreadCopyMoreGemmKPack = true;
+    GemmAThreadCopyMoreGemmK     = true;
+    do
+    {
+        if(!NextTwoPower<4, 256>(GemmMPerBlock))
+            break;
+        if(!NextTwoPower<16, 256>(GemmNPerBlock))
+            break;
+        if(!NextTwoPower<1, 8>(GemmKPerBlock))
+            break;
+        if(!NextTwoPower<4, 128>(GemmMPerWave))
+            break;
+        if(!NextTwoPower<16, 128>(GemmNPerWave))
+            break;
+        if(!NextTwoPower<1, 8>(GemmKPACKSize))
```

Comment:
@whchung Please let me know if this tuning range need adjustment. Thanks!

Reply:
@jerryyin please change KPACK range to just `<4, 8>`

Reply:
Done in 5511ba4